lib/authentication/SecureStorage/credentialManager.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2015-2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+const Logger = require('../../logger');
+const secureStorage = require('./secureStorage');
+const localStorage = require('./localStorage');
+
+/**
+ * 
+ * @param {String} host 
+ * @param {String} user 
+ * @param {String} credType 
+ * @returns 
+ */
+
+function checkForNull(...parameters){
+  return parameters.some((element) => element === undefined || element === null);
+}
+
+async function writeCredential(host, user, credType, token) {
+  if (checkForNull(host, user, credType, token)) {
+    Logger.getInstance().debug('Token cannot be saved. One of the paramenters is null or undefined');
+    return;
+  } else {
+    try {
+      try {
+        await secureStorage.writeCredential(host, user, credType, token);
```

Comment:
Check OS here and call secure/local storage accordingly.
https://stackoverflow.com/questions/8683895/how-do-i-determine-the-current-operating-system-with-node-js
No local storage on Win/Mac and if anything wrong we should fallback to use browser authentication.

Reply:
I fixed it.